src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeBuffer.cs

```diff
@@ -153,8 +153,6 @@ public void AcquirePointer(ref byte* pointer)
             if (_numBytes == Uninitialized)
                 throw NotInitialized();
 
-            pointer = null;
```

Comment:
This is changing behavior.  If DangerousAddRef throws, previously the pointer that was passed in would have been nulled out, but now it won't.  I think this should be reverted.

Reply:
(I opened an issue on Roslyn for this exact case almost two years ago: https://github.com/dotnet/roslyn/issues/42761)